services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/steps/SwitchSteps.java

```diff
@@ -63,15 +64,28 @@ public void verifySwitchesAmount(int expectedSwitchesAmount) {
         assertTrue(allSwitchesResponse.size() >= expectedSwitchesAmount);
     }
 
-    @Given("^select a random switch and alias it as '(.*)'$")
+    @Given("^select a switch and alias it as '(.*)'$")
     public void selectARandomSwitch(String switchAlias) {
         List<Switch> switches = getUnaliasedSwitches();
-        Random r = new Random();
-        Switch theSwitch = switches.get(r.nextInt(switches.size()));
-        log.info("Selected random switch with id: {}", theSwitch.getDpId());
+        Switch theSwitch = switches.get(0);
+        log.info("Selected switch with id: {}", theSwitch.getDpId());
         topologyUnderTest.addAlias(switchAlias, theSwitch);
     }
 
+    @Given("^select a switch with Openflow version '(.*)' and alias it as '(.*)'$")
+    public void selectARandomSwitchWithSpecificOfVersion(String ofVersion, String switchAlias) {
+        List<Switch> switches = getUnaliasedSwitches();
+
+        for (Switch s: switches) {
+            if (ofVersion.equalsIgnoreCase(s.getOfVersion())) {
+                log.info("Selected switch with id: {}", s.getDpId());
+                topologyUnderTest.addAlias(switchAlias, s);
+                return;
+            }
+        }
+        log.error("No switches found with Openflow version {}", ofVersion);
```

Comment:
You should throw instead of log, because we should always stop execution at this point.
You can also call `Assume.assumeTrue(false, "msg")` in order to mark scenario as 'skipped' instead of failed (if you want so)